docs/001_develop/03_client-capabilities/007_forms/002_smart-forms/01_json_schema.mdx

```diff
@@ -0,0 +1,317 @@
+---
+title: 'Foundation Forms - JSON schema'
+sidebar_label: 'JSON schema'
+id: foundation-forms-json-schema
+keywords: [genesis, foundation, ui, forms, smart forms]
+tags:
+  - genesis
+  - foundation
+  - ui
+  - forms
+  - smart forms
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import JsonSchemaFormExample from './examples/jsonSchema_example.js'
+
+# Foundation Forms JSON schema
+
+Foundation forms use a [JSON Schema](https://json-schema.org/understanding-json-schema/about) to define the underlying model of the form. This defines the objects and properties, as well as their types.
+
+If the source of the data is a resource in a [Data Server](/develop/server-capabilities/real-time-queries-data-server/) or [Event Handler](/develop/server-capabilities/core-business-logic-event-handler/) of your application, then you can use the JSON schema that is generated automatically by the Genesis Platform. Alternatively, you can define your own schema on the client.
+
+:::info
+This section covers aspects of the [Data Server](/develop/server-capabilities/real-time-queries-data-server/) and [DataBase](/develop/server-capabilities/data-model/) sections.
+The examples provided are simple, but it is worth exploring these sections in more depth.
+:::
+
+
+## Examples 
+
+### Server-side JSON schema
+
+Here is an example of a simple table defined in an application's **-tables-dictionary.kts** file. It is a simple table with three fields:
+
+**smart-forms-examples-tables-dictionary.kts**
+```kts
+ tables {
+    table(name = "SIMPLE_TRADE", id = 10_000) {
+        field("SIMPLE_TRADE_ID", STRING).sequence("ST")
+        field("QUANTITY", DOUBLE)
+        field("SIDE", STRING)
+        primaryKey("SIMPLE_TRADE_ID")
+    }
+    ...
+}
+```
+
+The data for this table is provided by a query in the application's **-dataserver.kts** file. This is the data source:
+
+**smart-forms-examples-dataserver.kts**
+```kts
+dataServer {
+    query("ALL_SIMPLE_TRADES", SIMPLE_TRADE)
+    ...
+}
+```
+
+The event that enables you to insert the data from the form into the database table is provided in the application's **-eventhandler.kts** file. This is the data source:
+
+**smart-forms-examples-eventhandler.kts**
+```kts
+eventHandler {
+    eventHandler<SimpleTrade>("SIMPLE_TRADE_INSERT", transactional = true) {
+        onCommit { event ->
+            val details = event.details
+            entityDb.insert(details)
+            ack()
+        }
+    }
+    ...
+}
+...
+```
+
+To insert the automatically-generated JSON schema for this resource in your **template** file, use the code below:
+
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
```

Comment:
I would like @submit-success and @submit-failure events added here?

When I re wrote the forms docs I didn't event know they existed - happy for it to be spun out to a separate PR

Reply:
Why here? All events are included and described in the DOM API section.

Reply:
Just to see an example of them in action.

Reply:
I can add it in separate PR